Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -632,13 +627,12 @@ function New-LDAPSIdentitySource {
     }
     # check if the certicates expire or not
     foreach ($cert in $Certificates) {
-        $certDate = Get-Date $cert.GetExpirationDateString()
         $currentDate = Get-Date
         Write-Host "Verifying certificate: $($cert.Subject)"
-        if ($certDate -lt $currentDate) {
-            Write-Error "The certificate is expired. The certificate is only valid not after $certDate." -ErrorAction Stop
+        if (($cert.NotBefore -lt $currentDate) -and ($cert.NotAfter -gt $currentDate)) {
+            Write-Host "The certificate is current."
         } else {
-            Write-Host "The certificate is valid."
+            Write-Error "The certificate is expired. The certificate is only valid not after $certDate." -ErrorAction Stop
```

Comment:
This message would be less confusing if we outlined the actual criteria we are testing for, something like: 
The certificate is not current, it's only valid between $start and $end dates.